inference-engine/src/auto_plugin/auto_exec_network.cpp

```diff
@@ -6,19 +6,187 @@
+            std::dynamic_pointer_cast<ngraph::op::ConvolutionIE>(node) ||
+            std::dynamic_pointer_cast<ngraph::op::DeconvolutionIE>(node)) {
+          auto layerType = node->input(1).get_element_type().get_type_name();
+          if (layerType == "f32")
+              return METRIC_VALUE(FP32);
+          if (layerType == "f16")
+              return METRIC_VALUE(FP16);
+        }
+    }
+    return METRIC_VALUE(FP32);
+}
+}  // namespace
+
+struct IdleGuard {
+    explicit IdleGuard(AutoExecutableNetwork::WorkerInferRequest* workerInferRequestPtr,
+                       AutoExecutableNetwork::NotBusyWorkerRequests& notBusyWorkerRequests) :
+        _workerInferRequestPtr{workerInferRequestPtr},
+        _notBusyWorkerRequests{&notBusyWorkerRequests} {
+    }
+    ~IdleGuard() {
+        if (nullptr != _notBusyWorkerRequests) {
+            _notBusyWorkerRequests->try_push(_workerInferRequestPtr);
+        }
+    }
+    AutoExecutableNetwork::NotBusyWorkerRequests* Release() {
+        auto notBusyWorkerRequests = _notBusyWorkerRequests;
+        _notBusyWorkerRequests = nullptr;
+        return notBusyWorkerRequests;
+    }
+    AutoExecutableNetwork::WorkerInferRequest*     _workerInferRequestPtr = nullptr;
+    AutoExecutableNetwork::NotBusyWorkerRequests*  _notBusyWorkerRequests = nullptr;
+};
+
+AutoExecutableNetwork::AutoExecutableNetwork(const std::string& modelPath,
+                                             const InferenceEngine::CNNNetwork& network,
+                                             const ConfigType& config,
+                                             AutoInferencePlugin* plugin)
+                                             : _autoPlugin(plugin) {
+    if (_autoPlugin->GetCore() == nullptr) {
+        IE_THROW() << "Please, work with AUTO device via InferencEngine::Core object";
+    }
+
+    if (modelPath.empty() && network.getFunction() == nullptr) {
+        IE_THROW() << "AUTO device supports just ngraph network representation";
+    }
+
+    auto metaDevices = _autoPlugin->GetDeviceList(config);
+    auto core = _autoPlugin->GetCore(); // shared_ptr that holds the Core while the lambda below (which captures that by val) works
+    auto LoadNetworkAsync =
+        [this, core, modelPath, network](std::vector<DeviceName> deviceList) -> IE::SoExecutableNetworkInternal {
+            DeviceName device;
+            IE::SoExecutableNetworkInternal executableNetwork;
+            auto networkPrecision = GetNetworkPrecision(network);
+            // select device
+            while (!deviceList.empty()) {
+                device = _autoPlugin->SelectDevice(deviceList, networkPrecision);
+                try {
+                    // std::cout << "!!! DEBUG: Starting Async loading to the " << device << " !!!" << std::endl;
+                    if (!modelPath.empty()) {
+                        executableNetwork = core->LoadNetwork(modelPath, device, {});
```

Comment:
BTW, why empty config is passed, how would THROUGHPUT/LETCY hints get expedited to the device?

Reply:
THROUGHPUT/LATENCY  hints are beyond this PR scop, the empty config is following the original code because AUTO will ignore device (like CPU) configuration #5979

Reply:
BTW, do you forget to submit other review comments?